### ISSUE TITLE
Have function like 'push' or 'replace' to manage navigation history?

### ISSUE THREAD

**Issue by DeFenceD** [react-community/react-navigation#2122]
I want to have some function like 'push', 'replace' for navigation history stack to decide which view can use 'back button' make the view go back.

**Comment by alexandref93** [comment]
You can create a push or replace function with NavigationActions

Example for push:

https://reactnavigation.org/docs/navigators/navigation-actions#Navigate

Example for reset:

https://reactnavigation.org/docs/navigators/navigation-actions#Reset

Snippet:
```javascript
import { NavigationActions } from 'react-navigation'

export function push (routeName, params) {
  return NavigationActions.navigate({
    routeName,
    params
  })
}

export function replace (routeName, params) {
  return NavigationActions.reset({
    index: 0,
    actions: [
      NavigationActions.navigate({ routeName, params})
    ]
  })
}

// Example 
this.props.navigation.dispatch(push('Settings'))
this.props.navigation.dispatch(replace('Profile'))

```

**Comment by DeFenceD** [comment]
OK, let me try again, and thank you very mach.

**Comment by Noitidart** [comment]
Thanks very much @DeFenceD and @alexandref93 for sharing this. I also needed a `replace` to avoid build up of some redux connected pages.

**Comment by Noitidart** [comment]
@alexandref93 - me being lazy here (short on multi-task-ability really) - do you have a snippet for discarding screen at index X?

**Comment by Noitidart** [comment]
@alexandref93 the pop out one of the previous entries in history without changing the page. Is this possible? My goal is this:


On routeName "landing" ->click on "?" -> Navigates to "help" -> click on "contact" -> Navigates to "contact", after navigation to "contact" complete, it should discard the previous page of "help" so that clicking "Back" while on "Contact" page should take me directly back to "landing".


Doing a wipe of all history, makes it so when it does go back to landing, that nothing is saved, and things start over like animations etc.

**Comment by Vitalik7** [comment]
```
import { NavigationActions, StackActions } from 'react-navigation'

export function push(routeName, params) {
  navigatorRef.dispatch(
    StackActions.push({
      routeName,
      params,
    })
  )
}
